source/Api.cpp

```diff
@@ -37,7 +39,9 @@ namespace Aws
             // sets up the StlAllocator for use.
             g_allocator = allocator;
             aws_mqtt_library_init(allocator);
-            aws_s3_library_init(allocator);
+            #ifdef ENABLE_S3_BUILT
+                aws_s3_library_init(allocator);
```

Comment:
`aws_s3_library_init` initializes several other libraries, including `aws_auth_library_init` and `aws_http_library_init`.

Reply:
That doesn't seem right :(